test/format/checksum.c

```diff
@@ -0,0 +1,136 @@
+#include "format.h"
+
+#define FNV_PRIME 0x00000100000001b3
+
+struct checksum_table_arg {
```

Comment:
should we create a seperate `checksum.h` file for this?

Reply:
Yeah, it's only intended to be used here. It's about 10 lines of actual logic for the hashing stuff, I'm sure if someone wants to use it elsewhere in future people can figure something out :-)